var/spack/repos/builtin/packages/aocl-sparse/package.py

```diff
@@ -64,18 +67,19 @@ def cmake(self, spec, prefix):
 
         args.extend([
             "-DBUILD_SHARED_LIBS:BOOL=%s" % (
-                'ON' if '+shared' in spec else 'OFF'
-            )
-        ])
+                'ON' if '+shared' in spec else 'OFF'),
 
-        args.extend([
             "-DBUILD_CLIENTS_BENCHMARKS:BOOL=%s" % (
-                'ON' if self.run_tests else 'OFF'
-            )
+                'ON' if self.run_tests else 'OFF')
         ])
 
-        with working_dir(self.build_directory, create=True):
-            inspect.getmodule(self).cmake(*args)
+        if "@3.0:" in spec:
+            args.extend([
+                "-DBUILD_ILP64:BOOL=%s" % (
+                    'ON' if '+ilp64' in spec else 'OFF')
+            ])
+
+        return args
 
     # Check that self.prefix is there after installation
     @run_after('build')
```

Comment:
FWIW. Regarding the comment above, `check` is technically testing the build -- after the `build` phase and using the `self.build_directory` -- not the installation.  Testing the installation would be done after the `install` phase and would be using the `self.prefix`, which is the root directory for the package's installed files.

Reply:
Removed following comment:
# Check that self.prefix is there after installation

Changed the following text from
Simple test to test the installation by running
to
Simple test to test the built library by running